cosima_cookbook/netcdf_utils.py

```diff
@@ -0,0 +1,8 @@
+def find_record_dimension(d):
```

Comment:
A netCDF-4 dataset may have multiple unlimited dimensions but this will only give one of them. Not a problem for any of the ACCESS-OM2 data though.

Reply:
Wow, I didn't realise that! Perhaps it would be better to look for something called `[Tt]ime`, or something with the `cartesian_axis` attribute set to `T`?

Reply:
Multiple unlimited dimensions is a `netCDF4` only feature. As our models still, by default, emit `netCFD3` it isn't a big deal for them. External data sources are another matter, and I agree looking for specific attributes might be a good idea in any case.

It is possible to cascade these checks. If there are multiple unlimited then require there to be some metadata to distinguish which is `Time`, in the case of single unlimited allow there to be no metadata (but maybe emit a small warning saying the assumption that is being made?). 

I guess it is possible for someone to make a file that has `Time` as a fixed dim and another axis as unlimited, in which case defer to the metadata? 

I can't find where this is used in the code BTW .. :)

Reply:
Used in `database.file_timeinfo()` (which in itself is fairly bodgy...)